resourseleak/TestResourcesWrappedByConstructorCallsClosed.java

```diff
@@ -0,0 +1,56 @@
+package resourceleak;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class TestResourcesWrappedByConstrucorCallsClosed {
+
+
+    protected static List<String> testResourcesWrappedByConstrucorCallsClosed(long timeout) throws IOException {
+
+        List<String> thisAndGitVersion= new ArrayList<String>();
+
+        // Get version of this ASCIIGenome
+        thisAndGitVersion.add(ArgParse.VERSION);
+
+        BufferedReader br= null;
+        timeout= timeout + System.currentTimeMillis();
+        // Get github versions
+        URL url = new URL("https://api.github.com/repos/dariober/ASCIIGenome/tags");
+        br = new BufferedReader(new InputStreamReader(url.openStream()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *br*. The resource is referenced in statements at the following lines: 30, 34. The resource closure statement is at line: 37. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Either a) close *br* in a try-finally block or b) close the resource by declaring *br* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).